### ISSUE TITLE
Ability to specify network tags for node pools in GKE cluster

### ISSUE THREAD

**Issue by snasovich** [rancher/rancher#33406]
**Request description:**
Add network tags when adding node pools to a GKE cluster during provisioning and edit.

**Actual behavior:**
Can't specify tags when working with node pools in a GKE cluster through Rancher.

**Expected behavior:**

- When creating a GKE v2 cluster through the Rancher UI, a user should be able to specify 0 or more network tags to add to the nodes in a node pool. Mimicking the behavior in the GCP console.
- When creating a GKE v2 cluster through the API ~or TF provider~ (separated into https://github.com/rancher/rancher/issues/34291), the user should be able to specify a list of 0 or more network tags.

**Additional notes:**
https://cloud.google.com/vpc/docs/add-remove-network-tags

_JIRA ID: SURE-3037_

**Comment by prabalsharma** [comment]
(tested on 9e0d1b2)

network tags created by operator are not visible in Rancher

<img width="1011" alt="Screen Shot 2021-07-23 at 12 41 27 PM" src="https://user-images.githubusercontent.com/3749546/126833201-1bc24407-a2ea-4575-8254-6ff7782abaec.png">


There is no network tag in upstream spec.

GCP:


<img width="927" alt="Screen Shot 2021-07-23 at 12 42 19 PM" src="https://user-images.githubusercontent.com/3749546/126833298-999f3014-996c-4352-a31c-8116dadd5f83.png">

**Comment by prabalsharma** [comment]
tested on master head

**Comment by sowmyav27** [comment]
Reopening to test from API/TF

**Comment by prabalsharma** [comment]
tested from API on v2.6-e22c6bb23ee7e65f04e953afb7a449532cb44db0-head

**Comment by prabalsharma** [comment]
looks like new argument needs to be added to tfp

```
± terraform apply

Error: Unsupported argument

  on terra.tf line 48, in resource "rancher2_cluster" "prabal-tf2":
  48:         tags = ["http-server", "https-server"]

An argument named "tags" is not expected here.
```

**Comment by prabalsharma** [comment]
tfp does not support rancher 2.6. WIP https://github.com/rancher/terraform-provider-rancher2/pull/711
Should be part of `Fix gke_config_v2 schema to rancher v2.6` task

**Comment by snasovich** [comment]
@prabalsharma , split out TF support into separate issue https://github.com/rancher/rancher/issues/34291 for more granular tracking. Let's track it there and close this one. Let me know if this needs to be handled differently.
